Stack/Opc.Ua.Core/Types/Encoders/XmlDecoder.cs

```diff
@@ -2745,7 +2745,7 @@ private Array ReadArrayElements(string fieldName, BuiltInType builtInType, Type
 
 
             // skip whitespace.
-            while (m_reader.NodeType != XmlNodeType.Element)
+            while (m_reader.NodeType != XmlNodeType.Element && m_reader.NodeType != XmlNodeType.EndElement)
             {
```

Comment:
Is this really needed? There is some concern this extra check can cause decoding issues.

Reply:
It tripped during testing but I can't reproduce anymore. I'll remove it and bring it back as a bug fix later if necessary.